src/graph.js

```diff
@@ -9,25 +15,35 @@ class Graph extends React.Component {
             branches: [],
             branchName:''
         };
+        this.componentRef = React.createRef();
     }
 
+    // Function to add commit from state
     addCommit = (branch) => {
 
         branch.commit(this.state[`commitMessage${branch.name}`]);
 
     };
 
+    //Function to add branch and adds branch name to branches array in state
     addBranch = () => {
+        if (this.state.branches
+            .map((b) => b.name)
+            .includes(this.state.branchName)
+        )
+                  return;
```

Comment:
Style nitpick: can you maybe improve the formatting a bit? The `return` looks a bit confusing (maybe put it in a bracket) and similarly for rest of the code. Maybe using Prettier would be helpful. :D

Also, I think you can make your functions 'const'.

Also, next time onwards, please make sure to create new branches for your PRs.

Reply:
I will keep that in mind